src/main/java/org/skife/jdbi/v2/sqlobject/ParameterBinderRegistry.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.skife.jdbi.v2.sqlobject;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+class ParameterBinderRegistry
+{
+    static final ParameterBinderFactory DEFAULT_BINDER_FACTORY = new ParameterBinderFactory() {
+        @Override
+        public boolean accepts(Class type, Method method, int param_idx) {
+            return true;
+        }
+
+        @Override
+        public Binder binderFor(Class type, Method method, int param_idx) {
+            return new PositionalBinder(param_idx);
+        }
```

Comment:
In v3, all binds are assumed to be by name even in the absence of a `@Bind` annotation, unless the sql object class was compiled without the `-parameters` compiler flag.

So there will have to be a discussion around positional bindings as the default when we get to v3.